stock_location_package_restriction/models/stock_move.py

```diff
@@ -1,109 +1,29 @@
         for location, move_lines in groupby(
             lines_being_processed, lambda m: m.location_dest_id
         ):
-            if not location.package_restriction:
-                continue
-
-            if new_location:
-                location = new_location
-            existing_package_ids = location_packages.get(location.id, set())
-            existing_packages = Package.browse(existing_package_ids).exists()
-            new_packages = Package.browse()
-            for move_line in move_lines:
-                package = move_line.result_package_id
-                # CASE: Package restiction exists but there is no package in move
-                if not package:
-                    error_msgs.append(
-                        _(
-                            "A package is mandatory on the location {location}. "
-                            "You cannot move the product {product} without a package."
-                        ).format(
-                            location=location.display_name,
-                            product=move_line.product_id.display_name,
-                        )
-                    )
-                    continue
-                if location.package_restriction != SINGLEPACKAGE:
-                    continue
-                # CASE: Package on location, new package is different
-                if existing_package_ids and package.id not in existing_package_ids:
-                    error_msgs.append(
-                        _(
-                            "Only one package is allowed on the location {location}."
-                            "You cannot add the {package}, there is already "
-                            "{existing_packages}."
-                        ).format(
-                            location=location.display_name,
-                            existing_packages=", ".join(
-                                existing_packages.mapped("name")
-                            ),
-                            package=package.display_name,
-                        )
-                    )
-                    continue
-                # CASE: Multiple packages in Move but single location
-                # with singlepackage Restriction
-                if new_packages and package not in new_packages:
-                    error_msgs.append(
-                        _(
-                            "Only one package is allowed on the location {location}."
-                            "You cannot move multiple packages to it:"
-                            "{packages}"
-                        ).format(
-                            location=location.display_name,
-                            packages=", ".join(new_packages.mapped("display_name")),
-                        )
-                    )
-                    continue
-                new_packages |= package
-
+            try:
+                with self.env.cr.savepoint():
+                    location._check_package_restriction(move_lines)
```

Comment:
Is the Savepoint needed?
There is a Union missing. Move lines is just a list.

```suggestion
                move_lines = self.env["srick.move.line"].union(*move_lines)
                location._check_package_restriction(move_lines)
```